simulation/adx_gyro_accelerometer/src/main/native/cpp/ADXRS450_SpiGyroWrapperData.cpp

```diff
@@ -38,6 +72,45 @@ void ADXRS450_SpiGyroWrapper::HandleRead(uint8_t* buffer, uint32_t count) {
   std::memcpy(&buffer[0], &returnCode, sizeof(returnCode));
 }
 
+void ADXRS450_SpiGyroWrapper::HandleAutoReceiveData(uint8_t* buffer,
+                                                    int32_t numToRead,
+                                                    int32_t& outputCount) {
+  std::lock_guard<wpi::mutex> lock(m_dataMutex);
+  double diff = m_angleDiff;
```

Comment:
You may be able to reduce the critical section if this read-modify-update doesn't need to occur atomically. Otherwise, if the mutex is locked for the entire function call, you could just use m_angleDiff instead of a copy.